chrome-extension/sidebar/sidebar.js

```diff
@@ -48,6 +48,318 @@ const htmlToElement = (html) => {
   return template.content.firstChild;
 };
 
+// Begin code for sorting course list
+
+document.addEventListener('change', function(event) {
+  if (event.target.matches('.size-dropdown')) {
+    handleDropdownChange(event);
+  }
+});
+
+// dropdown element
+const dropdownTemplate = `
+<div style="display: inline-block; margin-left: -20px; margin-right: 5px;">
+  <select class="size-dropdown" style="height: 34px; transform: translateY(3%); font-size: 14px; padding: 6px 12px;">
+    <option value="A">Sort By Course # (Default)</option>
+    <option value="B">Sort By Seats Available</option>
+    <option value="C">Sort By Most-Common Grade</option>
+    <option value="D">Sort By Popularity</option>
+    <option value="E">Sort By Units</option>
+  </select>
+</div>
+`;
+
+// handle sorting options
+const handleDropdownChange = (event) => {
+  if (event.target.value === "A") {
+    location.reload();
+  }
+  if (event.target.value === "B") {
+    const buttons = document.querySelectorAll('.action-sections.btn.btn-default');    
+    buttons.forEach(button => button.click());
+    setTimeout(sortBySeatsAvailable, 1000);
+  }
+  if (event.target.value === "C") {
+    sortByMostCommonGrade();
+  }
+  if (event.target.value === "D") {
+    sortByPopularity();
+  }
+  if (event.target.value === "E") {
+    sortByUnits();
+  }
+};
+
+const sortBySeatsAvailable = () => {
+  // Looks like { "CSCI4011":26, "CSCI4041":37 ... }
+  const courseSizeDict = {};
+
+  // Get the main container with class 'course-list-results'
+  let courseListResults = document.querySelector('.course-list-results');
+  let courseDivs = Array.from(courseListResults.firstElementChild.children);
+
+  for (let courseDiv of courseDivs) {
+    // Get name of course
+    let courseName = courseDiv.querySelector('a').getAttribute('name');
+    
+    // Get seats available value
+    let targetDiv = courseDiv.querySelector('div:nth-child(2) > div:nth-child(3)');
+    let trElements = targetDiv.querySelectorAll('tr');
+    
+    let sum = 0;
+    trElements.forEach(tr => {
+        let lastChildText = tr.lastElementChild.innerText;
+        sum += parseFloat(extractDifference(lastChildText)) || 0; // Ensure the text is converted to a number
+    });
+    
+    courseSizeDict[courseName] = sum;
+  }
+
+  // sort the courseDivs in the order we want
+  courseDivs.sort((a, b) => {
+    let nameA = a.querySelector('a').getAttribute('name');
+    let nameB = b.querySelector('a').getAttribute('name');
+    return courseSizeDict[nameB] - courseSizeDict[nameA];
+  });
+
+  console.log(courseSizeDict); // For debugging purposes, display the updated dictionary
```

Comment:
Add a prefix to these logs, like 

```
console.debug("[GG] sorting courses by size dict
", coursesSizeDict);
```

so that way if there's errors the variables are more descriptive and it's easier to understand what exactly is being printed.